From 798c81fe5525c3f7cd7b62645092543ab1989f4e Mon Sep 17 00:00:00 2001
From: Mike Yoder <myoder@cloudera.com>
Date: Tue, 29 May 2018 14:47:07 -0700
Subject: [PATCH 2823/2848] CLOUDERA-BUILD CDH-67049 Using jackson versions
 from the root pom

==C5_APPROVED_BUGFIX==

In this commit we use the versions of jackson that are supplied in the
root pom. Previously they were hard coded. This change needs to go in
before we bump up the version in the root pom, otherwise there are
packaging failures in hadoop.

This has been tested thanks to the efforts of Csaba Fulop, who created
a special set of jenkins jobs at

http://golden.jenkins.cloudera.com/job/CDH-MYODER5.16.0-Packaging-Hadoop/

specifically for this purpose. We have two green runs there.

(cherry picked from commit 514aeca4f89dd116266880d9b1682f162f869696)

Change-Id: Ic53c17fc599c2aa34f012d34b5bb17ff94579414
---
 .../cloudera/maven-packaging/hadoop-core/pom.xml   |   16 ++++------------
 hadoop-project/pom.xml                             |    4 ++--
 2 files changed, 6 insertions(+), 14 deletions(-)

diff --git a/hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-core/pom.xml b/hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-core/pom.xml
index 894a7ba..e68bd02 100644
--- a/hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-core/pom.xml
+++ b/hadoop-mapreduce1-project/cloudera/maven-packaging/hadoop-core/pom.xml
@@ -71,16 +71,6 @@
       <groupId>tomcat</groupId>
       <artifactId>jasper-compiler</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.codehaus.jackson</groupId>
-      <artifactId>jackson-core-asl</artifactId>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.codehaus.jackson</groupId>
-      <artifactId>jackson-mapper-asl</artifactId>
-      <scope>compile</scope>
-    </dependency>
     
     <!--<dependency>-->
     <!--<groupId>org.mortbay.jetty</groupId>-->
@@ -129,12 +119,14 @@
     <dependency>
       <groupId>org.codehaus.jackson</groupId>
       <artifactId>jackson-mapper-asl</artifactId>
-      <version>1.8.8</version>
+      <version>${cdh.jackson-mapper-asl.version}</version>
+      <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>org.codehaus.jackson</groupId>
       <artifactId>jackson-core-asl</artifactId>
-      <version>1.8.8</version>
+      <version>${cdh.jackson.version}</version>
+      <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.zookeeper</groupId>
diff --git a/hadoop-project/pom.xml b/hadoop-project/pom.xml
index d1bf277..1eccb62 100644
--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -65,7 +65,7 @@
     <jersey.version>1.9</jersey.version>
 
     <!-- jackson versions -->
-    <jackson.version>1.8.8</jackson.version>
+    <jackson.version>${cdh.jackson.version}</jackson.version>
     <jackson2.version>2.2.3</jackson2.version>
 
     <!-- ProtocolBuffer version, used to verify the protoc version and -->
@@ -754,7 +754,7 @@
       <dependency>
         <groupId>org.codehaus.jackson</groupId>
         <artifactId>jackson-mapper-asl</artifactId>
-        <version>${jackson.version}</version>
+        <version>${cdh.jackson-mapper-asl.version}</version>
       </dependency>
       <dependency>
         <groupId>org.codehaus.jackson</groupId>
-- 
1.7.9.5

